Add documentation for dataframe operator (#631)
astronomer/astro-sdk: docs/astro/sql/operators/dataframe.rst

> ==================

Reviewer: ```suggestion

.. dataframe_operator:

==================
```

> ======================================
> dataframe operator
> ======================================

Reviewer 2: ```suggestion
==================
dataframe operator
==================
```

.. _dataframe_operator:

> When to use the ``dataframe`` operator

Reviewer 3: There should be a mention in this section about how a user can use the `output_table` parameter to turn the resulting dataframe into a table.

Author: @dimberman `DataframeOperator` is inheriting from `DecoratedOperator` in the current [code](https://github.com/astronomer/astro-sdk/blob/main/src/astro/sql/operators/dataframe.py#L76). This doesn't have `output_table`.

I feel it should be inherited from `BaseSQLDecoratedOperator` as in [base_decorator](https://github.com/astronomer/astro-sdk/blob/main/src/astro/sql/operators/base_decorator.py#L15). 

WDYT?

Author: [Done](https://github.com/astronomer/astro-sdk/pull/631/commits/46ed25c9f06bf5a843281352718547b3dc5f47f2) the change @dimberman

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> ``dataframe`` decorator allows a user to run python functions in Airflow but with the huge benefit that SQL tables will automatically be turned into dataframes and resulting dataframes can automatically used in astro.sql functions. ``dataframe`` operator converts a SQL table into a dataframe. Users can give a python function that takes a dataframe as one of its inputs and run that python function. Once that function has completed, the result is accessible via the Taskflow API.

Reviewer 3: ```suggestion
``dataframe`` decorator allows a user to run python functions in Airflow but with the huge benefit that SQL tables will automatically be turned into dataframes and resulting dataframes can automatically used in astro.sql functions. When a ``Table`` object is passed into the``dataframe`` operator, the operator automatically converts the SQL table into a dataframe. Users can then give a python function that takes a dataframe as one of its inputs and run that python function against the rendered dataset. Once the python function has completed, the result is accessible via the Taskflow API.
```

> The ``dataframe`` decorator allows a user to run python functions in Airflow but with the huge benefit that SQL tables will automatically be turned into dataframes and resulting dataframes can automatically used in astro.sql functions. When a ``Table`` object is passed into the``dataframe`` operator, the operator automatically converts the SQL table into a dataframe. Users can then give a python function that takes a dataframe as one of its inputs and run that python function against the rendered dataset. Once the python function has completed, the result is accessible via the Taskflow API.

Reviewer 2: ```suggestion
The ``dataframe`` decorator allows a user to run python functions in Airflow but with the huge benefit that SQL tables will automatically be turned into dataframes and resulting dataframes can automatically used in astro.sql functions. When a ``Table`` object is passed into the ``dataframe`` operator, the operator automatically converts the SQL table into a dataframe. Users can then give a python function that takes a dataframe as one of its inputs and run that python function against the rendered dataset. Once the python function has completed, the result is accessible via the Taskflow API.
```

Example:

>     The following example aggregate the data and returns dataframe.

Reviewer 2: ```suggestion
    The following example aggregates the data and returns a dataframe.
```

> .. dataframe_operator:

Reviewer 4: ```suggestion
.. _dataframe_operator:
```

Author: [Done](https://github.com/astronomer/astro-sdk/pull/631/commits/46ed25c9f06bf5a843281352718547b3dc5f47f2)

==================
dataframe operator
==================

> .. _dataframe_operator:

Reviewer 4: Do you intend to have this same line on L2?

Reviewer 4: If so, we should remove this.

```suggestion
```

Author: [Done](https://github.com/astronomer/astro-sdk/pull/631/commits/46ed25c9f06bf5a843281352718547b3dc5f47f2)
